### PR TITLE
1. consoleappender中增加locale配置项的解析，方便输出中文

### PR DESCRIPTION
**Pull request by billgao87**

### CODE REVIEW

**Comment by wilx** [comment]
Please use English to report issues.

File: msvc14/log4cplus.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 14
-VisualStudioVersion = 14.0.22310.1
+VisualStudioVersion = 14.0.25420.1
```
Comment (wilx) [inline msvc14/log4cplus.sln]: Avoid committing changes in unrelated files.

File: include/log4cplus/consoleappender.h [hunk 2]
```diff
@@ -77,6 +77,8 @@ namespace log4cplus {
          * will be flushed at the end of each append operation.
          */
         bool immediateFlush;
+		
+		log4cplus::tstring localeName;
```
Comment (wilx) [inline include/log4cplus/consoleappender.h]: Use spaces, not tabs.

File: src/consoleappender.cxx [hunk 3]
```diff
@@ -27,6 +27,7 @@
 #include <log4cplus/spi/loggingevent.h>
 #include <log4cplus/thread/syncprims-pub-impl.h>
 #include <ostream>
+#include "log4cplus/spi/factory.h"
```
Comment (wilx) [inline src/consoleappender.cxx]: Use `<log4cplus/spi/factory.h>`, not `""`. Reorder together with other log4cplus headers.

File: src/consoleappender.cxx [hunk 4]
```diff
@@ -47,27 +48,54 @@ ConsoleAppender::getOutputMutex ()
     return helpers::getConsoleOutputMutex ();
 }
 
+static
+std::locale
+get_locale_by_name(tstring const & locale_name)
+{
+	try
```
Comment (wilx) [inline src/consoleappender.cxx]: Again, spaces vs tabs. Use spaces.

File: src/consoleappender.cxx [hunk 5]
```diff
@@ -47,27 +48,54 @@ ConsoleAppender::getOutputMutex ()
...
 // ConsoleAppender ctors and dtor
 //////////////////////////////////////////////////////////////////////////////
 
 ConsoleAppender::ConsoleAppender(bool logToStdErr_,
-    bool immediateFlush_)
-: logToStdErr(logToStdErr_),
-  immediateFlush(immediateFlush_)
+	bool immediateFlush_)
+	: logToStdErr(logToStdErr_),
+	immediateFlush(immediateFlush_),
```
Comment (wilx) [inline src/consoleappender.cxx]: Tabs vs spaces. Avoid reformatting unrelated lines.

File: src/consoleappender.cxx [hunk 6]
```diff
@@ -47,27 +48,54 @@ ConsoleAppender::getOutputMutex ()
     return helpers::getConsoleOutputMutex ();
 }
 
+static
+std::locale
+get_locale_by_name(tstring const & locale_name)
```
Comment (wilx) [inline src/consoleappender.cxx]: This function is clearly a copy of the same function from `fileappender.cxx`. Instead of duplicating the code, please move the function definition into `env.cxx` and declaration into `internal/env.h`.

File: src/consoleappender.cxx (hunk 4 above)
Comment (wilx) [reply src/consoleappender.cxx]: Also, please keep the formatting of `{try`.

**Comment by wilx** [comment]
Also, use commit messages only in English.

File: src/consoleappender.cxx [hunk 7]
```diff
@@ -103,7 +131,14 @@ ConsoleAppender::append(const spi::InternalLoggingEvent& event)
     thread::MutexGuard guard (getOutputMutex ());
 
     tostream& output = (logToStdErr ? tcerr : tcout);
+
+	auto cur_loc = output.getloc();
```
Comment (wilx) [inline src/consoleappender.cxx]: I think we want to make this conditional, only do this if user specifies different than default locale. Also, we should call flush before changing the locale and flush after we change the locale.

File: src/consoleappender.cxx (hunk 3 above)
Comment (billgao87) [reply src/consoleappender.cxx]: Did I just make a mistake?  I am not familiar with github, these changes only my test when i read the source code. What i wanted to do is to pull the new changes from mater to my branch, not to push my changes to source code! Please ignore the request of mine!

File: src/consoleappender.cxx [hunk 8]
```diff
@@ -47,27 +48,54 @@ ConsoleAppender::getOutputMutex ()
...
+	: Appender(properties),
+	logToStdErr(false),
+	immediateFlush(false),
+	localeName(LOG4CPLUS_TEXT("DEFAULT"))
 {
-    properties.getBool (logToStdErr, LOG4CPLUS_TEXT("logToStdErr"));
-    properties.getBool (immediateFlush, LOG4CPLUS_TEXT("ImmediateFlush"));
+	properties.getBool(logToStdErr, LOG4CPLUS_TEXT("logToStdErr"));
+	properties.getBool(immediateFlush, LOG4CPLUS_TEXT("ImmediateFlush"));
+	localeName = properties.getProperty(LOG4CPLUS_TEXT("Locale"), LOG4CPLUS_TEXT("DEFAULT"));
```
Comment (wilx) [inline src/consoleappender.cxx]: Instead of storing the locale name in the `ConsoleAppender` class, we should store the `std::locale` instance instead. But only if it is non-default locale. I think we will need a `std::unique_ptr<std::locale>` for that.

**Comment by wilx** [comment]
#569 was accepted instead of this.
